webapp/sources/rudder/rudder-core/src/main/resources/reportsSchema.sql

```diff
@@ -401,7 +401,7 @@ CREATE TABLE CampaignEvents (
   campaignId   text
 , eventid      text PRIMARY KEY
 , name         text
-, state        text
+, state        json
```

Comment:
I think we should keep a simple enumeration for that. And in fact, I find that having a case class for state makes everything much more complicated. I would prefer to keep the state and its informations/details in a separated columns.

In fact, we will likely need a "summary" or whatever the name columns, json, that give meta info about the state without having to make queries to other tables. That columns was in spec, but was not implemented yet. 
`
That column would be json, dependent of `state` and `campaignType` (perhaps). For `skipped`, it would give the reason. For `success`, a summary of things that worked ("5423 noes updates"). For error, some statistics about what happened. Etc. 

So, I will validate it like that now, but I would like to have it thought again before release